src/Microsoft.SqlTools.ServiceLayer/LanguageServices/ConnectedBindingContext.cs

```diff
@@ -162,55 +183,111 @@ public ParseOptions ParseOptions
+        private static DatabaseCompatibilityLevel GetDatabaseCompatibilityLevel(SMO.Server server)
         {
-            int versionMajor = Math.Max(serverVersion.Major, 8);
+            if (server.DatabaseEngineType == DatabaseEngineType.SqlAzureDatabase)
+            {
+                return DatabaseCompatibilityLevel.Azure;
+            }
 
-            switch (versionMajor)
+            // Get the actual compat level of the database we're connected to
+            switch (GetServerCompatabilityLevel(server))
             {
-                case 8:
+                case SMO.CompatibilityLevel.Version80:
                     return DatabaseCompatibilityLevel.Version80;
-                case 9:
+                case SMO.CompatibilityLevel.Version90:
                     return DatabaseCompatibilityLevel.Version90;
-                case 10:
+                case SMO.CompatibilityLevel.Version100:
                     return DatabaseCompatibilityLevel.Version100;
-                case 11:
+                case SMO.CompatibilityLevel.Version110:
                     return DatabaseCompatibilityLevel.Version110;
-                case 12:
+                case SMO.CompatibilityLevel.Version120:
                     return DatabaseCompatibilityLevel.Version120;
-                case 13:
+                case SMO.CompatibilityLevel.Version130:
                     return DatabaseCompatibilityLevel.Version130;
+                case SMO.CompatibilityLevel.Version140:
+                    return DatabaseCompatibilityLevel.Version140;
+                case SMO.CompatibilityLevel.Version150:
+                    return DatabaseCompatibilityLevel.Version150;
                 default:
                     return DatabaseCompatibilityLevel.Current;
             }
         }
 
         /// <summary>
-        /// Gets the transaction sql version from a server version
+        /// Gets the transaction sql version for a given server connection
         /// </summary>
-        /// <param name="serverVersion"></param>
-        private static TransactSqlVersion GetTransactSqlVersion(ServerVersion serverVersion)
+        /// <param name="server"></param>
+        private static TransactSqlVersion GetTransactSqlVersion(SMO.Server server)
         {
-            int versionMajor = Math.Max(serverVersion.Major, 9);
+            if (server.DatabaseEngineType == DatabaseEngineType.SqlAzureDatabase)
+            {
+                return TransactSqlVersion.Azure;
+            }
 
-            switch (versionMajor)
+            // Determine the language version to use - we can't just use VersionMajor directly because there are engine versions (such as MI)
+            // whose language version they support is higher than the actual server version. So we choose the highest compat level from
+            // between the server version and compat level
+            var compatLevel = Math.Max(server.VersionMajor * 10, (int)GetServerCompatabilityLevel(server));
```

Comment:
Compatibility

Reply:
🤦‍♂️ This is what happens when I go to an IDE that doesn't have a spellchecker...